### DISCUSSION TITLE
How to contribute to the repositories

### DISCUSSION THREAD

**Discussion by 0MAL** [mosparo/mosparo#204] [General]
Hi. I'm new to github. I found a mistake in translation. How can I contribute mosparo?
Can I just download, edit, and upload (overwrite) the yaml file on https://github.com/mosparo/mosparo/tree/master/translations?

**Comment by zepich** [comment]
Hi @0MAL 

Thank you very much for your question and your help in making mosparo better!

For translations, the easiest approach is to translate it in Weblate, so we directly get the change by pull request: https://hosted.weblate.org/projects/mosparo/

Here, you can see all the translations (the same as in the files in the `/translations` directory) and edit them in the language you want to adjust. The frontend translations (in the mosparo box in your form) are in the `Frontend` component in Weblate. The translations in the backend are in the `Backend-develop` and `Validators-develop` components in Weblate.

You can also fork the mosparo repository, adjust the content of the translation file in the YAML, and then send us a pull request. You can find more about that here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo and https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork

Please let me know if you need more information or help.

Thank you very much for your contribution!

Kind regards,

zepich

**Reply by 0MAL** [reply]
I managed to post pull request into github. Please check that.

**Reply by zepich** [reply]
@0MAL Perfect, thank you very much! I've merged it and we'll release a new version in the next days.

**Comment by 0MAL** [comment]
Hi, I have some questions.

Is it OK to submit a pull request without prior notice?

And I want to submit a pull request again to correct typos while the request #221 is closed. How can I resubmit a pull request to correct a pull request that is already closed?

Thanks.

**Reply by zepich** [reply]
Hi @0MAL 

Thank you very much for your questions.

Of course, you can send pull requests anytime and without telling us in advance. Thank you very much for your work!

You can submit a new pull request with additional changes. You can also create a pull request and save it as a draft, but wait to submit it to us (if you plan to make more changes).

I also want to inform you that we have made minor changes to the frontend translation file and added a new key for the next minor version of mosparo (v1.2). You can see the adjusted English translation file here: https://github.com/mosparo/mosparo/blob/develop/translations/frontend.en.yaml

Here you can see the changes: https://github.com/mosparo/mosparo/commit/b2f646cc8af135c81e9eaf989023377f3b25dce2#diff-87b50f850bb4d6df3882305f68c7b872f3147a419c2a9079920ae082e8614805

If you are working on the translations, you could already add these changes to the Japanese translation.

Kind regards,

zepich
